### PR TITLE
WFP-755 added grade to offender manager

### PR DESCRIPTION
**Pull request by carlov20**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/transformers/OffenderManagerTransformer.java [hunk 1]
```diff
@@ -111,4 +113,13 @@ private static boolean isUnallocated(final PrisonOffenderManager offenderManager
                 .orElse(false);
     }
 
+    private static OffenderManagerGrade offenderManagerGradeOf(final OffenderManager offenderManager) {
+        return Optional.ofNullable(offenderManager.getOfficer().getGrade())
+            .map(grade -> OffenderManagerGrade.builder()
+                .code(grade.getCodeValue())
+                .description(grade.getCodeDescription())
+                .build())
+            .orElse(null);
```
Comment (trozzomoj) [inline src/main/java/uk/gov/justice/digital/delius/transformers/OffenderManagerTransformer.java:122]: I do not really understand the use of orEsle(null) , that I can see is used everywhere , personally I believe is a bad practice and should be replaced with Optional.orElse()

**Review by trozzomoj — DISMISSED** [review]

File: src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/Officer.java [hunk 2]
```diff
@@ -36,4 +39,8 @@
 
     @Column(name = "FORENAME2")
     private String forename2;
+
+    @ManyToOne
+    @JoinColumn(name = "STAFF_GRADE_ID")
```
Comment (markreesmoj) [inline src/main/java/uk/gov/justice/digital/delius/jpa/standard/entity/Officer.java:44]: I am very scared of this DB
(fetch = FetchType.LAZY)
might be an idea here?
But ask someone who knows what they are talking about

File: src/main/java/uk/gov/justice/digital/delius/data/api/OffenderManagerGrade.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class OffenderManagerGrade {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/data/api/OffenderManagerGrade.java:13]: This works fine but there is a generic KeyValue class if you prefer to use that

File: src/testIntegration/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_GetAllOffenderManagersAPITest.java [hunk 4]
```diff
@@ -132,6 +135,9 @@ void canGetAllOffenderManagersByCrn() {
             assertThat(communityOffenderManager.getTeam()).isNotNull();
             assertThat(communityOffenderManager.getStaffCode()).isEqualTo("SH0007");
             assertThat(communityOffenderManager.getFromDate()).isEqualTo(LocalDate.of(2018, Month.MAY, 4));
+            assertThat(communityOffenderManager.getGrade()).isNotNull();
+            assertThat(communityOffenderManager.getGrade().getDescription()).isEqualTo("Accepted");
```
Comment (andymarke) [inline src/testIntegration/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_GetAllOffenderManagersAPITest.java:139]: Hmm, this looks like an odd grade, what has happened here?

**Review by andymarke — DISMISSED** [review]
LGTM

File: src/testIntegration/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_GetAllOffenderManagersAPITest.java (hunk 4 above)
Comment (carlov20) [reply src/testIntegration/java/uk/gov/justice/digital/delius/controller/secure/OffendersResource_GetAllOffenderManagersAPITest.java:139]: Its how the officer was created in the db https://github.com/ministryofjustice/community-api/blob/main/src/main/resources/db/data/V1_4__offenders_licences_data.sql#L250 where their STAFF_GRADE_ID was set to the reference number 1, happy to add in a script which sets the officer to an actual grade and assert on that?

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**